cpp/src/arrow/testing/gtest_common.h

```diff
@@ -45,7 +45,7 @@ class TestBase : public ::testing::Test {
   std::shared_ptr<Buffer> MakeRandomNullBitmap(int64_t length, int64_t null_count) {
     const int64_t null_nbytes = BitUtil::BytesForBits(length);
 
-    auto null_bitmap = *AllocateBuffer(null_nbytes, pool_);
+    std::shared_ptr<Buffer> null_bitmap = *AllocateBuffer(null_nbytes, pool_);
```

Comment:
Hmm... could you explain why this fixes the issue?

Reply:
@pitrou `auto null_bitmap` declares a `std::unique_ptr<Buffer>` which (being move-only) requires the implicit move-on-return (which is broken in gcc4.8)

Declaring a `std::shared_ptr<Buffer>` instead (which may be copied) allows the return statement to compile.